docs/components/material.md

```diff
@@ -327,42 +327,115 @@ visual defects.
 
 To work around this issue, try changing the order of the entities in the HTML.
 
-## Register a Custom Material
+## Register a Custom Shader Material
 
-We can register custom materials for appearances and effects using
-`AFRAME.registerShader`. However, the `registerShader` API is not yet fully
-featured and fairly limiting (e.g., no `tick` handler, some missing uniform
-types). For most cases for now, we recommend creating a custom material by
-creating three.js materials (e.g., `RawShaderMaterial`, `ShaderMaterial`) in a
-component:
+We can register custom shader materials for appearances and effects using
+`AFRAME.registerShader`.
+
+[example]: https://codepen.io/machenmusik/pen/WZyQNj
+Let's walk through an [example CodePen][example] with step-by-step commentary.
+
+As always, we need to include the A-Frame script.
+Here, we use RawGit to include the latest master version.
 
 ```js
-AFRAME.registerComponent('custom-material', {
+<script src="https://rawgit.com/aframevr/aframe/master/dist/aframe-master.min.js"></script>
+```
+
+Next, we define any custom components and shaders we may want to use,
+after A-Frame, but before the scene declaration.
+Here, we begin our `my-custom` shader.
+The schema declares any parameters for the shader.
+```js
+<script>
+AFRAME.registerShader('my-custom', {
   schema: {
-    // Add properties.
-  },
+```
+Where relevant, it is customary to support at least two data values:
 
-  init: function () {
-    this.material = this.el.getOrCreateObject3D('mesh').material = new THREE.ShaderMaterial({
-      // ...
-    });
+- `color`: A-Frame uses `type:'color'`, so it will parse string values like 'white' or 'red.
+- `opacity`: This is commonly used for fading in and out.
+
+`is:'uniform'` tells A-Frame this should appear as uniform value in the shader(s).
+
+```js
+    color: {type:'color', is:'uniform', default:'red'},
+    opacity: {type:'number', is:'uniform', default:1.0}
   },
+```
 
-  update: function () {
-    // Update `this.material`.
+Setting raw to true uses THREE.RawShaderMaterial instead of ShaderMaterial,
+so your shader strings are used as-is, for advanced shader usage.
```

Comment:
I'd link to the docs for [THREE.RawShaderMaterial](https://threejs.org/docs/#api/materials/RawShaderMaterial) and ShaderMaterial here.

> so your shader strings are used as-is, for advanced shader usage.

maybe better to be explicit: "so built-in uniforms and attributes are not automatically added to your shader code."

Reply:
done